### PR TITLE
fix: forward the API and content-gateway config to the browser engine

### LINKED ISSUE #970: Staging builds never receive VITE_WEB3AUTH_VERIFIER so login throws at session creation
Found while scoping the login verification steps for #888 (PR #969). Pre-existing and unrelated to that change.

## The gap

`createCoreKitSession` hard-requires both variables — `apps/web/src/auth/coreKit.ts:87-91`:

```ts
const clientId = env.VITE_WEB3AUTH_CLIENT_ID;
const verifier = env.VITE_WEB3AUTH_VERIFIER;
if (!clientId || !verifier) {
  throw new Error('VITE_WEB3AUTH_CLIENT_ID and VITE_WEB3AUTH_VERIFIER must both be configured');
}
```

`VITE_WEB3AUTH_CLIENT_ID` is plumbed through every workflow that builds the web app. `VITE_WEB3AUTH_VERIFIER` is plumbed through **none** — `git grep VITE_WEB3AUTH .github/workflows/` returns only `CLIENT_ID`:

- `.github/workflows/deploy-staging.yml` — 4 build sites, lines 102, 179, 272, 347
- `.github/workflows/desktop-staging-release.yml` — 3 build sites, lines 76, 162, 236
- `.github/workflows/web-e2e.yml` — lines 119, 125, 169

So every deployed staging build throws at session creation, and login cannot work.

It is silent at build time because `apps/web/src/vite-env.d.ts:12` declares the variable optional (`readonly VITE_WEB3AUTH_VERIFIER?: string`). Vite substitutes `undefined`, the bundle builds green, and the failure only appears in the browser at first login.

## Scope

- Add `VITE_WEB3AUTH_VERIFIER: ${{ vars.VITE_WEB3AUTH_VERIFIER }}` alongside every existing `VITE_WEB3AUTH_CLIENT_ID` in the three workflows above.
- **Set the `VITE_WEB3AUTH_VERIFIER` repository variable in GitHub settings.** The workflow edit alone changes nothing if the variable is unset — it would substitute empty and throw identically. This step is a repo-admin action and cannot be done from a PR.
- Consider whether a missing verifier should fail the build rather than the browser. A required-env assertion in the web build, or dropping the `?` in `vite-env.d.ts` and reading it through a checked accessor, would turn a silent green deploy into a red build. That is the reason this went unnoticed.

## Why it matters now

#888's acceptance needs a human to run a full staging login to confirm the new CSP does not break Web3Auth. Until this is fixed that login throws for an unrelated reason, and the failure will read as a CSP regression — the wrong diagnosis, on a security change.

## Depends on

Nothing in code. The repo-variable half depends on repo-admin access.

## Blocks

- #888 — its login verification cannot be performed on staging until this is fixed, and a failure would be misattributed to the CSP.
- #809 — the web login-and-CRUD e2e smoke gate drives a real login.
- #653 — the staging cutover assumes a working staging login.

Part of #655

### LINKED ISSUE #986: web: forward apiBaseUrl to the WASM EngineHandle so the browser engine logs into the API
`packages/client/src/worker/engineHost.ts:56-63` constructs `EngineHandle` with four config arguments hardcoded to `undefined`:

```ts
this.handle = new wasm.EngineHandle(
  seams, profile,
  undefined,  // api_base_url
  undefined,  // accelerator_base_url
  undefined,  // accelerator_bearer
  undefined,  // public_gateways
  storageHeadroomBytes
);
```

`apiBaseUrl` is already collected at `apps/web/src/engine/config.ts:49` and already reaches the JS Mailbox seam at `packages/client/src/browserSeams.ts:48`. It simply never reaches WASM: `packages/client/src/worker/engineWorker.ts:57` builds the host without threading it through.

## Consequence

`crates/engine/src/facade.rs:1315` guards on `if !self.api_base_url.is_empty()`. With an empty base that branch is false, so `start()` **skips `api.login_identity()` entirely**. The browser engine never authenticates to the API, and every publish, register, and upload runs against an empty base.

## Scope

- Thread `apiBaseUrl` from the worker config through `EngineHost` into `EngineHandle`.
- Decide and document what happens when it is absent — this is a trust-relevant path, so it should fail closed rather than silently skipping login.
- Cover it: assert the engine receives a non-empty base, and that `start()` performs the identity login.

## Why this is P0

Until this lands, #873, #807 and the #809 smoke gate are unverifiable even once written — the engine cannot authenticate, so no assertion about a real backend round-trip can pass.

Found during a critical-path audit on 2026-08-03 against `main` @ `bd30a0cbb`, verified against code rather than inferred from issue text. Companion to the content-gateway config gap filed alongside this one.

### LINKED ISSUE #987: web: add an accelerator and public gateway config surface for browser content reads
There is no configuration source anywhere in `apps/web` for a content accelerator or public IPFS gateway, so the browser engine has no way to fetch file content.

`crates/wasm/src/host.rs:160-162` states the intent directly:

> Dormant until the config slice supplies real endpoints: with no accelerator base URL and no fallbacks the gateway is empty, and reads fail closed as `Unavailable`.

`apps/web` defines five Vite variables — `VITE_ENVIRONMENT`, `VITE_ROUTING_ENDPOINTS`, `VITE_API_URL`, `VITE_WEB3AUTH_VERIFIER`, `VITE_WEB3AUTH_CLIENT_ID` — and none of them is an accelerator or gateway URL. `VITE_ROUTING_ENDPOINTS` feeds the **record** transport, not content retrieval.

## Consequence

No file content can be downloaded in the browser today, at any URL. Content reads fail closed as `Unavailable`, which is correct fail-closed behavior but leaves the read path non-functional.

## Scope

- Add an accelerator base URL and a public-gateway fallback list to the `apps/web` engine config surface.
- Thread both into `EngineHandle` alongside the `apiBaseUrl` gap tracked separately.
- Keep the fail-closed semantics: an unset gateway must stay `Unavailable`, never silently degrade to an untrusted default. The network is canonical and clients verify records cryptographically, so a fallback list is an accelerator hint, not a trust anchor.
- Decide whether `accelerator_bearer` belongs in this slice or is deferred, and record which.

## Why this is P0

#807 file download and static preview cannot function or be tested without it, and #809 cannot assert a content round-trip.

Found during a critical-path audit on 2026-08-03 against `main` @ `bd30a0cbb`, verified against code rather than inferred from issue text. Companion to the `apiBaseUrl` threading gap filed alongside this one.

### PR DESCRIPTION
**Pull request by FSM1**
Closes #986
Closes #987
Closes #970

Browser bootstrap and config wiring. No new primitives, no new KDF edges, no codec changes.

## #986 — apiBaseUrl reaches the WASM EngineHandle

`packages/client/src/worker/engineHost.ts` built `EngineHandle` with all four config arguments hardcoded `undefined`, so the engine ran against an empty API base and `Engine::start()` took its `if !self.api_base_url.is_empty()` branch — skipping `api.login_identity()` entirely. Every publish, register and upload ran unauthenticated against an empty base.

`apiBaseUrl` now flows `EngineHostConfig` to the worker bootstrap to `EngineHost` to `EngineHandle`. `EngineHost`'s positional arguments became an options object derived from `EngineHostConfig`, so the three layers carry one shape rather than three restatements, and `spawnEngineWorker` posts a checked spread instead of a hand-written field-by-field copy that nothing typechecked.

**An absent base fails closed at the WASM boundary**, rather than silently skipping login: `EngineHandle::new` returns a `JsError` for a missing, empty, or whitespace-only `apiBaseUrl`, so a misconfigured host cannot bring an engine up unauthenticated. The engine facade keeps its empty-base skip for the Rust harness — see "Deferred" below.

## #987 — accelerator and public gateway config surface

Adds `VITE_READ_ACCELERATOR_URL` and `VITE_PUBLIC_GATEWAYS` to the `apps/web` engine config, threaded into `GatewayConfig`. **No default**: unset leaves the gateway dormant and reads fail closed as `Unavailable`, never degrading to an endpoint nobody chose. The network is canonical and every fetched block is CID-verified in `crates/engine/src/content/read.rs`, so these are accelerator hints, not trust anchors.

Both are wired to repository variables in the staging web-build step; unset substitutes blank and stays dormant, so nothing changes until an admin sets them.

**`acceleratorBearer` is deferred, deliberately.** A `VITE_`-prefixed variable is inlined into the public browser bundle, and the accelerator token is a per-session credential. TypeScript passes `undefined`; #1033 tracks sourcing it from the session at runtime.

## #970 — VITE_WEB3AUTH_VERIFIER reaches every build

Plumbed alongside every existing `VITE_WEB3AUTH_CLIENT_ID` — 4 sites in `deploy-staging.yml`, 3 in `desktop-staging-release.yml`, 2 plus the `.env` write in `web-e2e.yml`.

The reason it went unnoticed is that a missing variable is silent at build time, so this also adds a build-time gate: a build naming a deployed environment that lacks a login-critical variable now fails the build instead of shipping a bundle that throws in the browser at first login. `VITE_API_URL` is gated too — an unset one falls back to `http://localhost:3000`, which the engine would then authenticate against.

The predicate lives in `apps/web/src/engine/config.ts` next to the other config-edge validators and is unit-tested; `vite.config.ts` and `createCoreKitSession` read the same list, so the gate and the runtime check cannot drift.

## Action required before this deploys

**Set the `VITE_WEB3AUTH_VERIFIER` repository variable.** This is a repo-admin action that cannot be done from a PR. Without it the staging deploy now goes **red at build time** rather than green-and-broken — the intended behaviour, but it will block a staging deploy until the variable exists.

Optionally set `VITE_READ_ACCELERATOR_URL` and `VITE_PUBLIC_GATEWAYS`; until then staging ships with a dormant content gateway and every content read fails closed as unavailable.

## Verification

Automated:

- `cargo fmt --all`, `cargo clippy --all-targets` clean, `cargo test -p cipherbox-engine` green across 13 suites.
- `cargo check` and `cargo test -p cipherbox-wasm --target wasm32-unknown-unknown` — 16 tests, including a new one asserting the constructor refuses an absent, empty, or whitespace-only `apiBaseUrl` before it touches the seam bag.
- `pnpm typecheck`, `pnpm lint`, `packages/client` 316 tests, `apps/web` 107 tests, `typecheck:browser-test`.
- Client Browser Suite, 30 Playwright tests, green. The engine worker now passes a real base, so cold start performs the identity login against a new in-memory mock of `/auth/challenge` and `/auth/login`. The mock refuses a login that does not echo a challenge it issued, and the cold-start test asserts the exchange tally moved — so a regression reinstating the login skip fails there rather than passing silently.
- The build gate was exercised three ways: a staging build missing the verifier fails, one missing `VITE_API_URL` fails, and an unconfigured `vite build` — the `Web Bundle` CI job — stays green.

Puppeteer against `vite dev`:

- With no login variables the login page renders `VITE_WEB3AUTH_CLIENT_ID and VITE_WEB3AUTH_VERIFIER must be configured`, reproducing #970's symptom.
- With both set the banner is gone and the login form renders clean, no console errors.

Needs a human:

- A full staging login once the repository variable is set. That is the half of #970 no PR can verify — it needs real Web3Auth credentials against the deployed bundle. Steps: set the repo variable, deploy staging, load the app, confirm no `must be configured` banner, then complete a Google login and an email login and confirm the vault loads.
- A content read against a real accelerator, once the gateway variables are set.

## Review gates

`/simplify`, `/security-review` and `/crypto-privacy-review` all run; findings folded in as a second commit.

The security pass found no HIGH or MEDIUM issues. It confirmed the verifier is a public routing identifier rather than a credential — sourced from `vars.`, exactly like the client id already in the bundle — that no path can source a bearer from build-time config, that a hostile gateway cannot forge content because every block is CID-verified with a mismatch terminal as a trust violation, and that the browser mock does not weaken any gate the Rust and contract suites already cover.

Folded in from the gates: `VITE_API_URL` added to the deploy gate; the login variable list single-sourced; the accelerator bearer's `Zeroizing` wrap moved above the constructor's first fallible step so an early return cannot drop it unzeroized; a whitespace-only API base trimmed; the bootstrap handshake given back its type annotation; the browser suite's login assertion made explicit; comments trimmed where they restated a rule stated elsewhere or defended an absent path.

## Deferred

- #1032 — make a blank API base unrepresentable in `Engine::new` rather than relying on one host's guard. Both the altitude and crypto passes wanted this; it is an `Engine::new` signature change across 9 call sites in a file with concurrent in-flight work, and the browser path it hardens is already closed at the host edge.
- #1033 — source the accelerator bearer from the session.
- #1034 — reject a login challenge that lacks the identity-login prefix before signing it. Pre-existing in `crates/engine/src/api/client.rs`, which this PR does not touch.

All three carry a native `blocked_by` edge to the issue that unblocked them.

## Merge sequencing

`crates/engine/src/facade.rs` is shared with in-flight work. This PR touches only two comment regions there — the `api_base_url` field doc and one stale line above the cold-start login guard — and changes no signature in that file. Every signature change is confined to `crates/wasm/src/host.rs`, whose seven-argument constructor already existed and only changed how it handles `api_base_url`, and to `packages/client`.

## Summary by CodeRabbit

* **New Features**
  * Added support for configuring read accelerators and public content gateways.
  * Improved authentication configuration for web and desktop builds.
  * Engine startup now accepts and forwards API, gateway, accelerator, and storage settings.

* **Bug Fixes**
  * Builds now clearly report missing required deployment or authentication settings.
  * Invalid or blank API URLs are rejected instead of silently defaulting.

* **Tests**
  * Expanded coverage for authentication, engine startup, gateway configuration, and login flows.

### CODE REVIEW

File: apps/web/src/engine/config.ts [hunk 1]
```diff
@@ -48,6 +69,35 @@ export function engineHostConfig(
...
+
+/**
+ * The variables a deployed bundle is missing. The bundler checks this so an
+ * unset one is a red build rather than a broken deploy nobody notices until a
+ * user tries to log in; a working-copy or CI build names no deployment and is
+ * exempt.
+ */
+export function missingDeployEnv(env: Partial<ImportMetaEnv>): string[] {
+  return DEPLOYED.includes(environment(env)) ? DEPLOY_ENV.filter((name) => !env[name]) : [];
+}
```
[the remark this answers is not in the document]
Comment (FSM1) [reply apps/web/src/engine/config.ts:122]: Accepted — confirmed on the branch before fixing. Wrote the assertions first and got 7 failures across every site you named: `missingDeployEnv` returned `[]` for a staging env with whitespace in all three variables, `acceleratorBaseUrl` came back as `'   '` instead of `undefined`, and `loginEnv` returned `{ clientId: '   ' }`.

The accelerator half was the sharpest, as you say: `'   '` builds a `GatewaySource` with an unusable `base_url`, so every read pays a doomed accelerator request before falling through to the public gateways, instead of the gateway staying dormant — and dormant is the fail-closed state this PR deliberately chose.

Fixed with one helper backing all three call sites, so absent, empty and whitespace-only are a single state:

```ts
function configured(value: string | undefined): string | undefined {
  return value?.trim() || undefined;
}
```

Used by `missingLoginEnv`, `missingDeployEnv` and `acceleratorBaseUrl`. `loginEnv` now returns the trimmed identifiers too, since they are sent to Web3Auth verbatim and a repo variable with a trailing newline would otherwise become part of the client id.

One deliberate deviation worth flagging, because the obvious version of this fix is a regression. I did **not** route `apiBaseUrl()` through `configured()`. That function defaults a blank value to `http://localhost:3000`, so trimming into the default would turn `VITE_API_URL="   "` from a hard refusal into a silent localhost fallback — exactly the hole you are reporting, one layer down. It now trims a *set* value but never defaults a whitespace one:

```ts
return env.VITE_API_URL === undefined || env.VITE_API_URL === ''
  ? DEFAULT_API_URL
  : env.VITE_API_URL.trim();
```

So `'   '` becomes `''`, and `EngineHandle::new` in `crates/wasm/src/host.rs` refuses it — that engine-edge check is untouched and still covers `None`, `Some("")` and `Some("  ")`. The two layers now agree: the build gate turns a broken deploy red before it ships, and the engine edge refuses it at runtime if one ever gets through.

Covered by runtime assertions in `config.test.ts` for whitespace-only accelerator, API URL, client id and verifier, plus a case pinning that a whitespace API origin does not default to localhost. `pnpm exec vite build` with `VITE_ENVIRONMENT=staging` now fails on a whitespace verifier and on a whitespace API URL, and still passes when both are real.

File: packages/client/test/browser/mockAuth.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+type Fields = Record<string, unknown>;
+
+function parse(body: Buffer): Fields {
+  try {
+    return JSON.parse(body.toString('utf8')) as Fields;
+  } catch {
+    return {};
+  }
```
[the remark this answers is not in the document]
Comment (FSM1) [reply packages/client/test/browser/mockAuth.ts:79]: Accepted, and reproduced exactly as described. `JSON.parse('null')` is the only input that reaches `field()` with a non-indexable value — `123`, `"str"`, `[]` and a parse error all return `null` harmlessly, because property access on a primitive boxes rather than throws. `null` throws inside the `.then` success callback, and since the promise is `void`-discarded the rejection is unhandled and the response is never written, so the request hangs.

Applied your fix:

```ts
let parsed: unknown;
try {
  parsed = JSON.parse(body.toString('utf8'));
} catch {
  return {};
}
return typeof parsed === 'object' && parsed !== null ? (parsed as Fields) : {};
```

Also added the runtime test you asked for, in the browser suite where this mock actually runs — it posts `null` to both `/auth/challenge` and `/auth/login` and expects 400. I verified it earns its keep by reverting the parse guard and re-running it: without the fix it fails on a request timeout rather than an assertion, which is precisely the "no useful signal" failure this is worth preventing. With the fix, both routes answer 400 and the suite is 31/31.

I considered a blanket `.catch()` around the dispatch so any future handler throw still answers, and left it out — it risks a double `send` if something ever throws after the response is written, and rejecting non-objects at the parse boundary closes the actual class of input. Kept the targeted fix.
